cpp/src/arrow/compute/kernels/vector_sort.cc

```diff
@@ -267,14 +267,19 @@ uint64_t* PartitionNulls(uint64_t* indices_begin, uint64_t* indices_end,
 // We need to preserve the options
 using PartitionNthToIndicesState = internal::OptionsWrapper<PartitionNthOptions>;
 
+// Convert string array to binary array and large string array to
+// large binary array. Because we want to reuse kernels for binary
+// array and large binary array to reduce generated codes.
```

Comment:
Can we do the same for temporal types?

Reply:
Ah, yes. We can reuse `Int*` code for temporal types.
I didn't notice it.

ARROW-10851